### ISSUE TITLE
Consider increasing the priority value of PrivateOption

### ISSUE THREAD

**Issue by MatiXOfficial** [pmateusz/meatie#56]
Currently, it is set to 50 while retry has 75. This means that `authenticate` is executed only once during retry.

Some exchanges, for example [binance](https://binance-docs.github.io/apidocs/spot/en/#signed-trade-user_data-and-margin-endpoint-security), require passing the current timestamp as a parameter of each request or including the timestamp in the request signature. The timestamp/signature should be regenerated when retrying. Since `authenticate` is called only at the beginning in the current meatie version, the timestamp and signature stay the same during retry attempts, which means the retry usually fails due to a too-old timestamp used.

**Comment by pmateusz** [comment]
Thank you @MatiXOfficial, I didn't consider extending `PrivateOption`, I changed the `priority` to be a property.

In the case you mentioned, it is also sensible to move evaluation of the `authenticate` option after the `limit`. The latter could also delay dispatching of the request.

The suggestion was implemented in the PR: [fix/56/move-authentication-after-retry](https://github.com/pmateusz/meatie/pull/58)
